### PR TITLE
Hydrate bug fix

### PR DESCRIPTION
**Pull request by prateekbh**
**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**
no

**Summary**
Makes sure hydrate is only used for correct URL

**Does this PR introduce a breaking change?**
no

### CODE REVIEW

File: packages/cli/lib/lib/entry.js [hunk 1]
```diff
@@ -43,19 +45,21 @@ if (typeof app === 'function') {
...
-			process.env.NODE_ENV !== 'production' ||
-			!hydrate
-				? render
-				: hydrate;
+		const currentURL = preRenderData.url ? normalizeURL(preRenderData.url) : '';
+		const canHydrate =
+			process.env.PRERENDER &&
+			process.env.NODE_ENV === 'production' &&
+			hydrate &&
+			currentURL === location.pathname;
```
Comment (developit) [inline packages/cli/lib/lib/entry.js]: wondering if we need to also include `location.search` here? (tbh I think we could make a case both ways)

```suggestion
			currentURL === location.pathname + location.search;
```

Comment (reznord) [reply packages/cli/lib/lib/entry.js]: makes sense, or else folks users might complain saying that query params aren't being retained. 😅

Comment (reznord) [reply packages/cli/lib/lib/entry.js]: Pushed the tiny change 😬

Comment (developit) [reply packages/cli/lib/lib/entry.js]: We may want to check both?  I'm worry about this:

`https://preactjs.com/?lang=en` <-- should match the content for `"/"` but now won't.

**Pull request merged**